com.unity.ml-agents/Runtime/Analytics/Events.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using Unity.MLAgents.Actuators;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents.Analytics
+{
+    internal struct InferenceEvent
+    {
+        public string BehaviorName;
+        public string BarracudaModelSource;
+        public string BarracudaModelVersion;
+        public string BarracudaModelProducer;
+        public string BarracudaPackageVersion;
+        public int InferenceDevice;
+        public List<EventObservationSpec> ObservationSpecs;
+        public EventActionSpec ActionSpec;
+        public int MemorySize;
+        public long TotalWeightSizeBytes;
+        public string ModelHash;
+    }
+
+    /// <summary>
+    /// Simplified version of ActionSpec struct for use in analytics
+    /// </summary>
+    [Serializable]
+    internal struct EventActionSpec
+    {
+        public int NumContinuousActions;
+        public int NumDiscreteActions;
+        public int[] BranchSizes;
+
+        public static EventActionSpec FromActionSpec(ActionSpec actionSpec)
+        {
+            var branchSizes = actionSpec.BranchSizes ?? Array.Empty<int>();
+            return new EventActionSpec
+            {
+                NumContinuousActions = actionSpec.NumContinuousActions,
+                NumDiscreteActions = actionSpec.NumDiscreteActions,
+                BranchSizes = branchSizes,
+            };
+        }
+    }
+
+    /// <summary>
+    /// Information about one dimension of an observation.
+    /// </summary>
+    [Serializable]
+    internal struct EventObservationDimensionInfo
```

Comment:
@vincentpierre Following up on the previous comment, I converted `int[] Shape` field into this for `EventObservationSpec`. We could also keep using the `int[] Shape` and add an `int[] Flags` - that seems a little nicer, but this way ensures that the arrays are the same size, and it's cleaner if we ever need more per-dimension data.

Reply:
I think I prefer this option yes

Reply:
(confirmed offline that "this" refers to the code as is, not the comment)